assert/assertions.go

```diff
@@ -174,6 +174,17 @@ func ObjectsAreEqualValues(expected, actual interface{}) bool {
 
 	expectedType := expectedValue.Type()
 	actualType := actualValue.Type()
+
+	if expectedType.Kind() == actualType.Kind() {
```

Comment:
I think in the spirit of EqualValues, array and slice should also pass:
```go
assert.EqualValues(t, []interface{}{1}, [1]interface{}{int64(1)})
```

But this comparison prevents it.